### PR TITLE
Add prometheus definitions for key metrics.

### PR DESCRIPTION
**Pull request by mkcp**
Bumps go.md to the latest go-metrics to include the prometheus definitions patch and updates vendor. Also defines various key metrics for operating consul for their use in the beta in `lib/telemetry.go`.

### CODE REVIEW

File: vendor/github.com/armon/go-metrics/prometheus/prometheus.go [hunk 1]
```diff
@@ -39,19 +60,44 @@ type PrometheusSink struct {
...
 
-type PrometheusSummary struct {
+type summary struct {
 	prometheus.Summary
 	updatedAt time.Time
+	canDelete bool
+}
+
+// CounterDefinition can be provided to PrometheusOpts to declare a constant counter that is not deleted on expiry.
+type CounterDefinition struct {
```
Comment (s-christoff) [inline vendor/github.com/armon/go-metrics/prometheus/prometheus.go:91]: Does this need to `fmt`

**Review by s-christoff — APPROVED** [review]
One fmt question but otherwise looks great. 
🤙

File: vendor/github.com/armon/go-metrics/prometheus/prometheus.go (hunk 1 above)
Comment (mkcp) [reply vendor/github.com/armon/go-metrics/prometheus/prometheus.go:91]: The Go fmt check did not bark at this file when I merged it upstream, and we're not able to change it here without a patch. I have it run as a post-write hook in my editor, so it should be good!

**Comment by mikemorris** [comment]
Added a changelog entry in https://github.com/hashicorp/consul/pull/9129

**Pull request merged**